ElasticsearchReportHelper.php

```diff
@@ -561,6 +561,110 @@ public static function permissionFilters(array $options) {
+
+    // Add My records options.
+    $optionArr['p-my'] = lang::get('My records');
+
+    // Add All records if website permission allows.
+    if (!empty($options['all_records_permission']) && hostsite_user_has_permission($options['all_records_permission'])) {
+      $optionArr['p-all'] = lang::get('All records');
+    }
+
+    // Add collated location (e.g. LRC boundary) records if website permissions allow.
+    if (!empty($options['location_collation_records_permission'])
+        && hostsite_user_has_permission($options['location_collation_records_permission'])) {
+      $locationId = hostsite_get_user_field('location_collation');
+      if ($locationId) {
+        $locationData = data_entry_helper::get_population_data([
+          'table' => 'location',
+          'extraParams' => $options['readAuth'] + ['id' => $locationId],
+        ]);
+        if (count($locationData) > 0) {
+          $optionArr['p-location_collation'] = lang::get('Records within location ' . $locationData[0]['name']);
+        }
+      }
+    }
+
+    // Add in permission filters.
+    foreach ($sharingCodes as $sharingCode) {
+      $filterData = report_filters_load_existing($options['readAuth'], $sharingCode, TRUE);
+      foreach ($filterData as $filter) {
+        if ($filter['defines_permissions'] === 't') {
+          // If useSharingPrefix options specified, prefix type of sharing to filter name
+          $filterTitle = $options['useSharingPrefix'] ? $sharingTypes[$sharingCode].' - '.$filter['title'] : $filter['title'];
+          $optionArr['f-'.$filter['id']] = $filterTitle;
+        }
+      }
+    }
+
+    // Group page integration if user linked to warehouse.
+    $params = array(
+      'user_id'=>hostsite_get_user_field('indicia_user_id'),
+      'view' => 'detail'
+    );
+
+    if ($params['user_id']) {
+      $params['sharing']= 'reporting'; //RJB added this for testing from non-iRecord site
+
+      $groups = data_entry_helper::get_population_data(array(
+        'table'=>'groups_user',
+        'extraParams'=>data_entry_helper::$js_read_tokens + $params
+      ));
+      foreach ($groups as $group) {
+        $title = $group['group_title'] .
+            (isset($group['group_expired']) && $group['group_expired'] === 't' ? ' (' . lang::get('finished') . ')' : '');
+        if ($group['administrator']==='t') {
+          $optionArr["g-all-$group[group_id]"] = lang::get('All records added using a recording form for {1}', $title);
+        }
+        $optionArr["g-my-$group[group_id]"] = lang::get('My records added using a recording form for {1}', $title);
+      }
+    }
+
+    // Return the select control. There will always be at least one option (my records)
+    $controlOptions = [
+      'label' => lang::get('Record context'),
+      'fieldname' => $options['id'],
+      'lookupValues' => $optionArr,
+      'data' => $options['linkedUserFilter'],
+      'class' => 'record-context',
+    ];
+
+    helper_base::$javascript .= <<<JS
```

Comment:
Does the linked user filter need to be configured? Can you not just find user filters by class? Actually I think this is an unlikely scenario - if the recordContext (or permissionsFilter) control is on the page managing filters where defines_permissions=true, it's highly likely that userFilters would only be used to manage filters where defines_permissions=false. So you might not need to bother with this link?

Reply:
I noticed an unecessary line (656) just before the block you highlighted which was left-over from a mistaken approach I had taken, so I will remove that anyway. Yes I could find the userFilters control by class - I was accounting for situations where more than one used, but I guess this is unlikely to offer any advantage. I included disabling userFilter when permission filters/group contexts selected to mirror the setup on easy download 2, but it is not essential. I can remove it for now and add in later if there is any demand for it.

Reply:
I've removed the link.